tests/integration/utils.py

```diff
@@ -134,6 +134,15 @@ def run_cmd_on_image(
                 )
         return super().run_cmd_on_image(cmd, tmp_path, mounts, net)
 
+    def __exit__(self, exc_type: Any, exc_value: Any, exc_traceback: Any) -> None:
+        """Override the context manager exit to remove any dangling layers as build side effect."""
+        remove_dangling = "podman rmi --force $(podman images -q --filter 'dangling=true')"
```

Comment:
This would remove all dangling images in the user's filesystem, not just the ones generated by this test run, right? I'm just wondering if this could have an unintended side-effect, since Cachi2 is potentially deleting things it did not generate.

Reply:
> This would remove all dangling images in the user's filesystem, not just the ones generated by this test run, right? 

Yes, it would. But because these are always marked as `<none> <none>` you can't really tell without deep inspection on what generated those, so they're practically meaningless and it's actually podman's issue that it didn't delete them despite being instructed. Worst that happens is that future unrelated builds would get a bit slower due to missing layers from the intermediate images which nobody actually cares about in multi-stage builds, so this is a safe change from that perspective.

Reply:
> you can't really tell without deep inspection on what generated those

and I might even be wrong about this, because I simply didn't feel like tracing all the hashes on my local file system, so I **only suspect** that it's even possible trace them, but can't tell for sure.

Reply:
I saw that podman-images has a `after/since` filter:

```
 Filter by images created after the given IMAGE (name or tag).     
```

Which we could maybe leverage? But you're probably right it's not a big deal to delete them all, so I'm thinking we shouldn't waste time investigating.

Reply:
> I saw that podman-images has a after/since filter:

IIUC in order to execute this the way you suggest we'd have to remember the last built image as a reference for the removal. However, there could be parallel builds happening in the meantime in which case you'd still delete unrelated dangling images.

> so I'm thinking we shouldn't waste time investigating.

Agreed, I mean they're called dangling for a reason :) .